lib/services/cordova-project-plugins-service.ts

```diff
@@ -465,11 +507,27 @@ export class CordovaProjectPluginsService implements IPluginsService {
 
 			return condition;
 		});
+		let realIdentifier: string;
+		if(!matchingPlugins || !matchingPlugins.length) {
+			realIdentifier = pluginName;
+		}
+
+		realIdentifier = realIdentifier || matchingPlugins[0].data.Identifier;
+		let obsoletedBy = this.getObsoletedByPluginIdentifier(realIdentifier).wait();
+		if(obsoletedBy) {
+			matchingPlugins = matchingPlugins.concat(_.filter(plugins, (_plugin: IPlugin) => {
```

Comment:
This code is a bit hard to read, consider splitting it. Not a merge stopper, though.